.github/workflows/version_tests.yaml

```diff
@@ -0,0 +1,121 @@
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - uses: actions/checkout@v2
+        if: github.event_name != 'pull_request'
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-  uses: docker/setup-buildx-action@v1
+      - name: Build Docker image
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          file: Dockerfile
+          context: .
+          push: false
+          tags: k8ssandra/k8ssandra-operator:latest
+          platforms: linux/amd64
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache
+          outputs: type=docker,dest=/tmp/k8ssandra-k8ssandra-operator.tar
+      - name: Upload k8ssandra-operator image
+        uses: actions/upload-artifact@v2
+        with:
+          name: k8ssandra-operator
+          path: /tmp/k8ssandra-k8ssandra-operator.tar
+  run_kuttl_test:
+    runs-on: ubuntu-latest
+    needs: [build_image]
+    name: Run kuttl tests
+    strategy:
+      matrix:
+        k8s_version: ["1.21"]
+        kuttl-test:
+        - test-cassandra-versions
+        serverVersion: 
+        - 3.11.10
+        - 3.11.11
+        - 3.11.12
+        - 4.0.0
+        - 4.0.1
+        - 4.0.2
+        - 4.0.3
+    steps:
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+      if: github.event_name == 'pull_request'
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+    - uses: actions/checkout@v2
+      if: github.event_name != 'pull_request'
+    - name: Set up Go 1.16
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.16
+    - uses: actions/cache@v2
+      with:
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+    - name: Cache Docker layers
+      uses: actions/cache@v2
+      with:
+        path: /tmp/.buildx-cache
+        key: ${{ runner.os }}-buildx-${{ github.sha }}
+        restore-keys: |
+          ${{ runner.os }}-buildx-  uses: docker/setup-buildx-action@v1
+    - name: Kind kube-proxy issue workaround
+      run: sudo sysctl net/netfilter/nf_conntrack_max=524288
+    - name: Download k8ssandra-operator image
+      uses: actions/download-artifact@v2
+      with:
+        name: k8ssandra-operator
+        path: /tmp
+    - name: Load Docker images
+      run: |
+        docker load --input /tmp/k8ssandra-k8ssandra-operator.tar
+    - name: install kustomize
+      run: |
+        make kustomize
+    - name: Run kuttl-test
+      run: |
+        # We are running tests against k8s 1.20 - 1.22 currently. 
```

Comment:
Aren't we testing just 1.21?

Reply:
Yes we are. That comment was left over from a copy + paste from another workflow. Let me fix that...